src/main/java/org/aion/wallet/dto/AccountDTO.java

```diff
@@ -1,30 +1,35 @@
 package org.aion.wallet.dto;
 
 import org.aion.base.util.TypeConverter;
+import org.aion.wallet.connector.dto.BlockDTO;
+import org.aion.wallet.connector.dto.TransactionDTO;
 import org.aion.wallet.util.QRCodeUtils;
 
 import java.awt.image.BufferedImage;
-import java.util.Objects;
+import java.util.*;
 
 public class AccountDTO {
     private final String currency;
     private final String publicAddress;
+    private final boolean isImported;
+    private final int derivationIndex;
+    private final BufferedImage qrCode;
+    private final SortedSet<TransactionDTO> transactions;
     private byte[] privateKey;
     private String balance;  //TODO this has to be BigInteger
     private String name;
     private boolean active;
-    private final boolean isImported;
-    private final int derivationIndex;
-    private final BufferedImage qrCode;
+    private BlockDTO lastCheckedBlock = null;
 
-    public AccountDTO(final String name, final String publicAddress, final String balance, final String currency, boolean isImported, int derivationIndex) {
+    public AccountDTO(final String name, final String publicAddress, final String balance, final String currency, boolean isImported, int derivationIndex, SortedSet<TransactionDTO> transactions) {
         this.name = name;
         this.publicAddress = TypeConverter.toJsonHex(publicAddress);
         this.balance = balance;
         this.currency = currency;
         this.qrCode = QRCodeUtils.writeQRCode(publicAddress);
         this.isImported = isImported;
         this.derivationIndex = derivationIndex;
+        this.transactions = transactions;
```

Comment:
Do we need to pass the Tx list as a param? I think it would be ok if `this.transactions` would be initialized by default as a `new TreeSet(TRANSACTION_COMPARATOR)` and not passed from the outside.